homeassistant/components/nest/__init__.py

```diff
@@ -223,6 +238,50 @@ async def async_config_reload() -> None:
     return True
 
 
+async def async_import_config(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Attempt to import configuration.yaml settings."""
+    config = hass.data[DOMAIN][DATA_NEST_CONFIG]
+    new_data = {
+        **entry.data,
+        CONF_PROJECT_ID: config[CONF_PROJECT_ID],
+    }
+    if CONF_SUBSCRIBER_ID not in entry.data:
+        if CONF_SUBSCRIBER_ID not in config:
+            raise ValueError("Configuration option 'subscriber_id' missing")
+        new_data.update(
+            {
+                CONF_SUBSCRIBER_ID: config[CONF_SUBSCRIBER_ID],
+                CONF_SUBSCRIBER_ID_IMPORTED: True,  # Don't delete user managed subscriber
+            }
+        )
+    hass.config_entries.async_update_entry(entry, data=new_data)
+
+    if entry.data["auth_implementation"] == INSTALLED_AUTH_DOMAIN:
+        _LOGGER.warning(
+            "Nest has deprecated App Auth and OAuth credentials must be "
+            "re-created by you, following the latest integration "
+            "instructions. Please delete the integration, remove nest from "
```

Comment:
I have another related scenario where i'm making a decision but want confirmation before I go too far.

Scenario: A user has an existing `configuration.yaml`  and no config entry (e.g. set they set up w/ old instructions either web or app auth and didn't complete, or they deleted the existing entry). Because they have no config entry, none of the import logic runs.

Now, say they start a new config flow. I think the options are:
1) Act like the configuration.yaml doesn't exist at all it uses the `missing_credentials` flow with app credentials entered in the UI
2) Support signup from `configuration.yaml` but present a dialog to choose between app auth or web auth since we don't know what it is,
3) Support signup from web auth only via `configuration.yaml` and add more dialogs explaining the situation and rules, etc.

I'm going with option #1 as its simplest and is a reasonable balance between constraints of user simplicity and # of states to document.

Reply:
Yeah, I think 1. is ok in this case.

Reply:
As i've started getting this working, one snag is:
- Start re-auth flow
- Reauth flow starts, then progresses and aborts with `missing_credentials`
- User enters application credentials
- Now, frontend can't resume the reauth flow because it was aborted.

I'm playing around with starting re-auth again within the config flow as a task after aborting, then having the frontend somehow learn about the new flow, but haven't totally figured it out yet.

Reply:
How do we handle missing credentials when starting a new user flow?

Reply:
The config flow aborts, then the frontend takes user to the dialog, then when done the frontend restarts a new config flow for that integration.

(I also prototyped an alternative that was purely core based using config flows to accept app creds and write to storage, not sharing any code with the app creds UI dialogs, so that's a possibility. It works.)

Reply:
Could the frontend start a new flow with reauth context on missing credentials, ie always start a new flow with the same context as the aborted flow?

Reply:
It looks like it starts reauth with an existing flow id today. I'll look into starting a new flow with that same context. I imagine that needs new apis.

I am also looking at having core.abort and start a task to trigger reauth again, then the frontend can ask: have any on progress flows?  However that only works if there's one without some context matching.

Reply:
OK I got something working!  Though it has a lot of little steps, so next want to checkpoint here, and get feedback on ways to iterate and improve this.

We have an existing app auth credential that enters reauth state:

<img width="641" alt="Reauth step 1" src="https://user-images.githubusercontent.com/6026418/173247607-e5b3bc6a-6133-4ba3-ab9f-2109953f55f0.png">

More on this re-auth step later:

<img width="445" alt="Reauth step 2" src="https://user-images.githubusercontent.com/6026418/173247611-5efcfe0e-d2a1-4035-bc79-42723b6a0530.png">

<img width="603" alt="Reauth step 3" src="https://user-images.githubusercontent.com/6026418/173247613-9d2433c9-07d3-4e8f-adad-8ff92fc3777c.png">

The flow is aborted with `missing_credentials` triggering this page in the frontend and user is redirected to app credentials dialog. Behind the scenes, note there is a hack in core that schedules the config entry to re-entry the reauth state while replying with the aborted state:

<img width="600" alt="Reauth step 4" src="https://user-images.githubusercontent.com/6026418/173247617-e6e9127e-300a-4ecb-8744-cbb268d53072.png">

This is the application credentials with description placeholders populated:

<img width="497" alt="Reauth step 5" src="https://user-images.githubusercontent.com/6026418/173247772-5c20c53a-6abd-4214-8226-02ee376ccce4.png">

User enters the creds, and the frontend restarts a new config flow.  The frontend can see the re-auth state and prompts the user to select it explicitly:

<img width="471" alt="Reauth step 6" src="https://user-images.githubusercontent.com/6026418/173247618-c9ee970d-7446-4266-b20f-4d66f66f61c6.png">

Second time shown this re-auth dialog (more on this later):

<img width="442" alt="Reauth step 7" src="https://user-images.githubusercontent.com/6026418/173247779-3e421b48-299b-4caa-998d-f917b00fb41c.png">

User is shown a new extra step after "pick implementation" but before actually redirecting asks them to upgrade their device access project and reminds them of the OAuth ID to link in the cloud console:

<img width="598" alt="Reauth step 8" src="https://user-images.githubusercontent.com/6026418/173247627-9fe19f9c-77a8-403b-93df-daec74c09e8d.png">



The main complexity and extra screens here are because the need to restart the flow. I added back the extra reauth confirm step because the config entry is placed into re-auth before the the new credentials are created. That extra step needs to check the current state of things before proceeding (hence the "Nest needs to re-authenticate your account" in both passes -- it needs to make sure you actually created credentials).  The other step related to this is re-selecting the same config entry again after the creds. There *may* be a way to pick automatically with a frontend change, which I can look into next.

Anyway, taking a short checkpoint / pause for additional thoughts.

Reply:
Sounds ok. It would be nice if we could avoid duplicating the re-auth confirmation dialog, but not a blocking problem I think.